common/src/parse.cpp

```diff
@@ -149,29 +149,42 @@ pcl::console::parse_argument (int argc, const char * const * argv, const char *
 
 ////////////////////////////////////////////////////////////////////////////////
 std::vector<int>
-pcl::console::parse_file_extension_argument (int argc, const char * const * argv, const std::string &extension)
+pcl::console::parse_file_extension_argument (int argc, const char * const * argv,
+  const std::vector<std::string> &extension)
 {
   std::vector<int> indices;
   for (int i = 1; i < argc; ++i)
   {
     std::string fname = std::string (argv[i]);
-    std::string ext = extension;
+    bool found = false;
+    for (int j = 0; j < extension.size (); ++j)
+    {
+      std::string ext = extension[j];
 
-    // Needs to be at least 4: .ext
-    if (fname.size () <= 4)
-      continue;
+      // Needs to be at least 4: .ext
+      if (fname.size () <= 4)
+        continue;
 
-    // For being case insensitive
-    std::transform (fname.begin (), fname.end (), fname.begin (), tolower);
-    std::transform (ext.begin (), ext.end (), ext.begin (), tolower);
+      // For being case insensitive
+      std::transform (fname.begin (), fname.end (), fname.begin (), tolower);
+      std::transform (ext.begin (), ext.end (), ext.begin (), tolower);
 
-    // Check if found
-    std::string::size_type it;
-    if ((it = fname.rfind (ext)) != std::string::npos)
+      // Check if found
+      std::string::size_type it;
+      if ((it = fname.rfind (ext)) != std::string::npos)
+      {
+        // Additional check: we want to be able to differentiate between .p and .png
+        if ((ext.size () - (fname.size () - it)) == 0)
+        {
+          indices.push_back (i);
+          found = true;
+        }
+      }
+    }
+
+    if (found)
     {
-      // Additional check: we want to be able to differentiate between .p and .png
-      if ((ext.size () - (fname.size () - it)) == 0)
-        indices.push_back (i);
+      continue;
```

Comment:
I don't understand what you are trying to achieve here. `continue` in the end of the iteration does not do anything, it will "continue" anyway.

Reply:
Sorry, it was a pretty bad joke (but makes me laugth a lot).